GVRf/Framework/framework/src/main/jni/objects/components/render_data.h

```diff
@@ -51,13 +64,56 @@ class RenderData: public Component {
     };
 
     RenderData() :
-            Component(RenderData::getComponentType()), mesh_(0), light_(0), use_light_(false), use_lightmap_(false),
-                    render_mask_(DEFAULT_RENDER_MASK), rendering_order_(
-                    DEFAULT_RENDERING_ORDER), offset_(false), offset_factor_(
+            Component(RenderData::getComponentType()), mesh_(0), light_(0), use_light_(false), use_lightmap_(false),batching_(false),
+                    render_mask_(DEFAULT_RENDER_MASK), batch_(nullptr),rendering_order_(
+                    DEFAULT_RENDERING_ORDER), hash_code_dirty_(true), offset_(false), offset_factor_(
                     0.0f), offset_units_(0.0f), depth_test_(true), alpha_blend_(
                     true), alpha_to_coverage_(false), sample_coverage_(1.0f), invert_coverage_mask_(GL_FALSE), draw_mode_(GL_TRIANGLES), texture_capturer(0) {
     }
 
+    void copy(RenderData& rdata){
```

Comment:
Please make the parameter rdata const. Then, you can use this function to impl. the copy constructor without duplicating the code.